File: frontend/src/services/authService.js
import apiClient from './apiClient';

export const authService = {
  async login(credentials) {
    const response = await apiClient.post('/auth/login', credentials);
    return response.data?.data;
  },

  async register(userData) {
    const response = await apiClient.post('/auth/register', userData);
    return response.data?.data;
  },

  async getCurrentUser() {
    const response = await apiClient.get('/auth/me');
    return response.data?.data;
  },

  async resetPassword(data) {
    const response = await apiClient.post('/auth/reset-password', data);
    return response.data?.data;
  },

  async deleteUser(username) {
    const response = await apiClient.delete(`/auth/user/${encodeURIComponent(username)}`);
    return response.data;
  },
};

File: backend/src/main/java/com/yourorg/appname/service/AuthService.java
package com.yourorg.appname.service;

import com.yourorg.appname.dto.request.LoginRequest;
import com.yourorg.appname.dto.request.RegisterRequest;
import com.yourorg.appname.dto.request.ResetPasswordRequest;
import com.yourorg.appname.dto.response.AuthResponse;
import com.yourorg.appname.dto.response.UserResponse;

public interface AuthService {

    AuthResponse login(LoginRequest request);

    AuthResponse register(RegisterRequest request);

    UserResponse getCurrentUser();

    AuthResponse resetPassword(ResetPasswordRequest request);

    void deleteUser(String username);
}

